### DISCUSSION TITLE
Storage node example in kube.tf

### DISCUSSION THREAD

**Discussion by loxy** [mysticaltech/terraform-hcloud-kube-hetzner#1351] [Q&A]
In the `kube.tf` example there is an example of a storage node

```tf
{
      name        = "storage",
      server_type = "cpx21",
      location    = "fsn1",
      # Fully optional, just a demo.
      labels      = [
        "node.kubernetes.io/server-usage=storage"
      ],
      taints      = [],
      count       = 1
}
```

Can someone please explain for what I need that? Is it only relevant when using for Lonhorn?

**Comment by WebSpider** [comment]
No, labels can be used for scheduling workloads onto nodes, and work together with taints and tolerations. 
There is a lot out there on the subject, and ChatGPT can probably help you as well, but in short:

Labels:
- Are applied to a node
- Give hints as to what this node can be used for
Taints:
- Are applied to a node
- Can cause a pod not to be scheduled on a node, even if it has a label would cause it to normally be selected
Tolerations:
- Are applied to a pod
- Can make scheduling that would fail because of a Taint still happen

In practise, this is used to make sure the right node gets the right workload, so it *could* apply to Longhorn storage nodes as well, but doesn't need to.

**Reply by loxy** [reply]
Thank you for you answer! But I already asked ChatGPT a lot ;-)

I know what labels and taints are. My question was more in the direction of reason to create/user such a node config in general. It is "enabled" by default in `kube.tf` and creates a dedicated node (for storage as the name suggests). But I don't need that when using cloud volumes, right? 

That's why I wanted to understand why someone would use that node (because agents where already defined before). So, I guess no pods should be scheduled to it?

When it is not only used for Longhorn, can you give other examples where this setup is useful and I should use a "storage" node?

Sorry for my maybe dumb questions...
